### DISCUSSION TITLE
Ability to Remove xsi types

### DISCUSSION THREAD

**Discussion by micthom** [tefra/xsdata#981] [Q&A]
Using the XML Serializer I am able to generate XML files however it is adding "xsi" datatypes to a lot of the elements and I would prefer not to reduce size of the XML. If I look at the XML I need to send to the system it does not have these repeated  xsi:type="sometype" added. Is there a way to turn these off from generating?

For example:
    <device xsi:type="device:Device">
      <device:device_ID xsi:type="cndc:EntityIdentifierType">
        <cndc:resource>test</cndc:resource>
      </device:device_ID>
      <device:configuration xsi:type="device:Configuration"/>
    <device>

**Comment by tefra** [comment]
The xsi:type attribute is necessary when you are assigning derived elements, otherwise the produced xml documents wont pass validations. If you think there is a bug please open an issue.  That being said feel free to use the [tree serializer](https://xsdata.readthedocs.io/en/latest/data_binding/tree_serializing/), I am sure with xlst you can apply any transformation you want.
